cmake/extensions.cmake

```diff
@@ -65,6 +65,27 @@
 #  zephyr_compile_options()     --->  target_compile_options()
 #
 
+# override the target_sources function to inject a unique ID definition for each source file
+function(target_sources target_name)
```

Comment:
we should really try to avoid overriding CMake's built-in functions.

It can make it harder for users to understand what is happening, and it's extremely easy to make mistakes.